release.md

```diff
@@ -1,30 +1,94 @@
+
+In this phase:
+
+ - A PR must be opened in our [test-infra](https://github.com/falcosecurity/test-infra/blob/master/config/config.yaml) repository to set the newly created branch as protected.
+
+ - A release candidate tag should be created in the release branch for testing purposes.
+
+ - Accurate testing is performed on the release candidate. It's highly recommended to use Falco as a consumer when testing libs and drivers.
+
+ - If necessary, bug fixing or testing PRs can be merged on the `master` branch. In such a case, relevant commits are cherry-picked and ported to the release branch and a new release candidate is created.
+
+### Thaw
+
+Once maintainers are trustful that the release candidate is in good shape, or after ~5 days from the code freeze, we enter the Thaw phase. 
 
->_Note that not all versioned releases will be built and distributed. Drivers distribution is indeed implemented and maintained only to satisfy Falco's needs._
+From a technical perspective, this means that now the `master` and release branches diverge. 
 
-The libs release process is currently under development and limited to the versioning process only. The [libs versioning](#Libs-versioning) process happens in this repository.
+Now, the release branch is git tagged with the targeted version.
 
+From this point on:
+- Only bug fix PRs are allowed to be merged in the release branch 
+   - if any, the version patch number must be bumped  to git tag the release branch again.
+- All kinds of PRs are allowed to be merged in the `master` branch again.
 
-## Release procedure
+## Exceptions
 
-Regardless if it is a driver or a libs release when initiating a new release, we do the following process:
+Exceptions are allowed for compelling reasons. Notably:
 
-1. We decide together (usually in the #falco channel in [slack](https://kubernetes.slack.com/messages/falco)) if the source code is in a shape good enough to be released
-2. We double-check if the versioning rules have been respected (see sections below), then we pick the next version number to tag (i.e., a _git tag_)
-3. A person with repository rights creates a [new release using the GitHub UI](https://github.com/falcosecurity/libs/releases/new) (a git tag will be automatically created)
+- During the code freeze phase, PRs to complete or disable features in the code may be exceptionally merged
+- Hotfixes releases can happen anytime and without following the full process. In such cases, patches are merged into the relevant release branch (or a release branch is created if needed), then a new version is git tagged, and the hotfix is directly released.
 
-> _At the time of writing, no other steps are needed since the whole release process is still in development. This document will be updated once the definitive process is fully implemented._
+Exceptions to the [versioning rules](#versioning) are never allowed.
 
+## Versioning
 
-## Drivers versioning
+### Drivers versioning
```

Comment:
Falco prints this version. `Default driver` equates to `Driver version` right?

```
./falco --version

Falco version: XXX
Libs version:  XXX
Plugin API:    1.0.0
Driver:
  API version:    2.0.0
  Schema version: 2.0.0
  Default driver: 2.0.0+driver
```
Would below or something similar perhaps be clearer @leogr ?

```
./falco --version

Falco version: XXX
Libs version:  XXX
Plugin API version: 1.0.0
Driver version: 2.0.0+driver
Driver internal API version: 2.0.0
Driver internal schema version: 2.0.0
```

Reply:
Anyway, `Default driver` is not the driver version. The default driver version is just the version that `falco-driver-loader` (shipped together with Falco) will install by default.

In reality, Falco will be compatible with a range of drivers. And the compatibility check only uses API and schema versions.

Perhaps we should write "shipped with driver version: 2.0.0+driver" or something similar? 

cc @falcosecurity/falco-maintainers

Reply:
Hah that is actually a nice explanation, could you find a nice place (probably more in the Falco repo) to explain this exactly this way? Given this new info consider the suggestion invalidated.